gallery/gallery-library/src/main/java/org/silverpeas/components/gallery/MediaUtil.java

```diff
@@ -212,138 +207,117 @@ public static void processPhoto(final FileHandler fileHandler, final Photo photo
    * @param watermarkOther the secondary metadata retrieved to compute the watermark
    * @throws Exception
    */
-  public static void processPhoto(final FileHandler fileHandler, final Photo photo,
+  public synchronized static void processPhoto(final FileHandler fileHandler, final Photo photo,
       final File image, final boolean watermark, final String watermarkHD,
       final String watermarkOther) throws Exception {
     if (image != null) {
       try {
         photo.setFileName(image.getName());
         final HandledFile handledImageFile = getHandledFile(fileHandler, photo);
         fileHandler.copyFile(image, handledImageFile);
-        if (setInternalMetadata(handledImageFile, photo, MediaMimeType.PHOTOS)) {
-          createPhoto(handledImageFile, photo, watermark, watermarkHD, watermarkOther);
-        }
+        new PhotoProcess(handledImageFile, photo, watermark, watermarkHD, watermarkOther).process();
       } finally {
         FileUtils.deleteQuietly(image);
       }
     }
   }
 
   /**
-   * Gets a handled file.
-   * @param fileHandler the current session file handler
-   * @param media the original media file to get.
-   * @return the handled file
+   * Pastes media from a source to a destination.
+   * @param fileHandler the file handler (space quota management).
+   * @param fromPK the source.
+   * @param media the destination.
+   * @param cut true if it is a cut operation, false if it is a copy one.
    */
-  private static HandledFile getHandledFile(FileHandler fileHandler, InternalMedia media) {
-    if (StringUtil.isNotDefined(media.getFileName())) {
-      throw new IllegalArgumentException("media.getFilename() must return a defined name");
-    }
-    return fileHandler.getHandledFile(Media.BASE_PATH, media.getComponentInstanceId(),
-        media.getWorkspaceSubFolderName(), media.getFileName());
-  }
+  public synchronized static void pasteInternalMedia(final FileHandler fileHandler,
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Reorder the modifiers to comply with the Java Language Specification. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AModifiersOrderCheck)